src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,140 @@ package io.github.yuutoproject.yuutobot.commands
 
 import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
 import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import io.github.yuutoproject.yuutobot.utils.EMOJI_REGEX
+import io.github.yuutoproject.yuutobot.utils.NONASCII_REGEX
+import io.github.yuutoproject.yuutobot.utils.httpClient
 import java.io.IOException
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.utils.data.DataObject
 import net.dv8tion.jda.internal.utils.IOUtil
 import okhttp3.*
 import okhttp3.Callback as OkHttp3Callback
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.slf4j.LoggerFactory
+
+class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "Generates an image of a character in Camp Buddy saying anything you want", "[bg] <char> <text>") {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    @Suppress("UNCHECKED_CAST")
+    private fun getBackgroundsAndCharacters(): Pair<List<String>, List<String>> {
+        val request = Request.Builder()
+            .url("https://yuuto.dunctebot.com/info")
+            .get()
+            .build()
+
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code != 200) {
+                throw Exception("Failed to sync backgrounds and characters with API")
+            }
+            val json = DataObject.fromJson(IOUtil.readFully(IOUtil.getBody(response)))
+            return Pair(
+                json.getArray("backgrounds").toList() as List<String>,
+                json.getArray("characters").toList() as List<String>
+            )
+        }
+    }
+
+    private val bgAndChars = getBackgroundsAndCharacters()
+    private val backgrounds = bgAndChars.first
+    private val characters = bgAndChars.second
+
+    private val backgroundsString = "`${backgrounds.joinToString("`, `")}`"
+    private val charactersString = "`${characters.joinToString("`, `")}`"
+
+    private val json = "application/json; charset=utf-8".toMediaType()
 
-// TODO: get arch to implement the actual command
-class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "does something", "[bg] <char> <text>") {
     override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
-        val moveThisToAStorageLocation = OkHttpClient()
+        val now = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
```

Comment:
Shouldn't we change this error message to `This command requires at least two arguments`?

```suggestion
            event.channel.sendMessage("This command requires at least two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
```